substrate/frame/aura/src/migrations.rs

```diff
@@ -43,3 +45,108 @@ pub fn remove_last_timestamp<T: RemoveLastTimestamp>() -> Weight {
 	LastTimestamp::<T>::kill();
 	T::DbWeight::get().writes(1)
 }
+
+/// Migrate `CurrentSlot` to account for a slot duration change.
+///
+/// This migration recalculates the current slot based on the current timestamp and the new
+/// slot duration. This is necessary when upgrading to a runtime with a different slot duration,
+/// as the slot number calculated from the timestamp using the new duration may be less than
+/// the previously stored slot, which would cause a panic in `on_initialize`.
+///
+/// The migration:
+/// 1. Gets the current timestamp from `pallet_timestamp`
+/// 2. Gets the new slot duration from the runtime configuration
+/// 3. Calculates the new slot: `timestamp / slot_duration`
+/// 4. Updates `CurrentSlot` with the recalculated value
+///
+/// This migration is safe to call multiple times (idempotent).
+pub struct MigrateCurrentSlot<T>(core::marker::PhantomData<T>);
```

Comment:
While pallet `on_runtime_upgrade` isn't the default anymore. I think it is totally fine here to directly put this into the `on_runtime_upgrade` hook of the pallet. This way users only need to change the slot duration and do not need to care about anything else.